ADALiOS/ADALiOS/ADURLProtocol.m

```diff
@@ -102,29 +119,66 @@ -(void) connection:(NSURLConnection *)connection
 
 #pragma mark - NSURLConnectionDataDelegate Methods
 
-- (NSURLRequest *)connection:(NSURLConnection *)connection willSendRequest:(NSURLRequest *)request redirectResponse:(NSURLResponse *)response
+- (NSURLRequest *)connection:(NSURLConnection *)connection willSendRequest:(NSURLRequest *)request redirectResponse:(NSURLResponse *)redirectResponse
 {
-    AD_LOG_VERBOSE_F(sLog, @"HTTPProtocol::connection:willSendRequest:. Redirect response: %@. New request:%@", response.URL, request.URL);
+    AD_LOG_VERBOSE_F(sLog, @"HTTPProtocol::connection:willSendRequest:. Redirect response: %@. New request:%@", redirectResponse.URL, request.URL);
     
     // Disallow HTTP for ADURLProtocol
     if ([request.URL.scheme isEqualToString:@"http"])
     {
+        if ([request.URL.absoluteString.lowercaseString hasPrefix:s_endURL])
+        {
+            // In this case we want to create an NSURLError so we can intercept the URL in the webview
+            // delegate, while still forcing the connection to cancel. This error is the same one the
+            // OS sends if it's unable to connect to the host
+            [connection cancel];
+            NSError* failingError = [NSError errorWithDomain:NSURLErrorDomain
+                                                        code:-1003
+                                                    userInfo:@{ NSURLErrorFailingURLErrorKey : request.URL }];
+            [self.client URLProtocol:self didFailWithError:failingError];
```

Comment:
I don't think so, we had to do that before because we weren't getting hit with shouldStartLoadRequest: I'll know for sure when I get a test pass done.